### PR TITLE
Saved application

### PR DESCRIPTION
**Pull request by dfaulken**
A copy of #274.

Thanks @AkivaGreen for your hard work on this to get it to this point!

### CODE REVIEW

**Comment by dfaulken** [comment]
Sorry about the assigning and unassigning, I have ape fingers today — @Anbranin if you could review, I will be working to close this out.

File: app/controllers/application_records_controller.rb [hunk 1]
```diff
@@ -75,13 +75,11 @@ def toggle_saved_for_later
...
       flash[:message] = 'Application moved back to dashboard.'
     else
-      date = if params[:date_for_later].present?
-               Date.strptime(params[:date_for_later], '%m/%d/%Y')
-             end
-      mail = true if params[:mail_to_applicant] == '1'
-      @record.save_for_later(date: date,
+      parameters = save_for_later_params
+      @record.save_for_later(date: parameters[:date],
                              note: params[:note_for_later],
```
Comment (dfaulken) [inline app/controllers/application_records_controller.rb]: That method interpolates params. No interpolation needed for the note for later.

File: app/models/application_record.rb [hunk 2]
```diff
@@ -137,4 +146,12 @@ def self.ethnicity_eeo_data(records)
       [ethnicity, specific_records.count, specific_records.interview_count]
     end
   end
+
+  def self.send_notification_emails!(records)
+    records_by_email = records.includes(:position).group_by(&:email_to_notify)
+    records_by_email.each_pair do |address, ars|
```
Comment (dfaulken) [inline app/models/application_record.rb]: Block variable.

File: app/controllers/application_records_controller.rb (hunk 1 above)
Comment (Anbranin) [inline app/controllers/application_records_controller.rb]: Ah okay.

**Pull request merged**